Newsletters: add 188 (2022-02-23) (#724)
bitcoinops/bitcoinops.github.io: _posts/en/newsletters/2022-02-23-newsletter.md

---
title: 'Bitcoin Optech Newsletter #188'
permalink: /en/newsletters/2022/02/23/
name: 2022-02-23-newsletter
slug: 2022-02-23-newsletter
type: newsletter
layout: newsletter
lang: en
---
This week's newsletter FIXME

## News

- **Fee bumping and transaction fee sponsorship:** separate from the
  replace-by-fee discussion started a couple weeks ago (see [Newsletter
  #186][news186 rbf]), this week James O'Beirne [started][obeirne bump]
  a discussion about fee bumping.  In particular, O'Beirne is concerned
  that some of the transaction relay policy changes being proposed will
  complicate the use of fee bumping for users and wallet developers.  As
  an alternative, he seeks renewed consideration of [transaction fee
  sponsorship][topic fee sponsorship] (previously described in
  [Newsletter #116][news116 sponsorship]).

>     The idea received significant discussion on the mailing list.

Reviewer: I appreciate the effort to be objective, I probably would have said "significant pushback"

few spare moments to help curious or confused users.  In
this monthly feature, we highlight some of the top-voted questions and
answers posted since our last update.*

{% comment %}<!-- https://bitcoin.stackexchange.com/search?tab=votes&q=created%3a1m..%20is%3aanswer -->{% endcomment %}
{% assign bse = "https://bitcoin.stackexchange.com/a/" %}

- [Will a post-subsidy block with no transactions include a coinbase transaction?]({{bse}}112193)
  Pieter Wuille explains that since every block must have a coinbase transaction
  and since every transaction must include at least one
  input and one output, a post-subsidy block with no block reward (no fees and
  no subsidy) will still require at least one zero-value output.

- [How can the genesis block contain arbitrary data on it if the script is invalid?]({{bse}}112439)
  Pieter Wuille lists the reasons why the genesis block's coinbase
  "Chancellor..." text push is valid. First, the [genesis block][bitcoin se 13122] is valid by
  definition. Second is that coinbase input scripts are not actually executed.
  Third is that, pre-[BIP342][], the requirement of a single element on the stack

>   after execution is only a policy rule, not consensus. Finally, that policy

Reviewer 2: ```suggestion
  after execution is only a policy rule, not a consensus rule. Finally, that policy
```

>   rule applies only when executing both input and output scripts and since the
>   genesis block's input does not have an output script, the policy does not
>   apply. Pieter also notes the reason for the genesis block's unspendability is

Reviewer 2: ```suggestion
  rule applies only to the final stack after an input script is executed together with the corresponding output script. Since there 
  are no corresponding output scripts for the inputs of coinbase transactions, the policy does not
  apply. Wuille also notes the reason for the genesis block's unspendability is
```

Reviewer 2: (we refer to people with their last name above, so I changed it here as well)

  unrelated to this discussion and involves the original Bitcoin software [not
  adding the genesis block][bitcoin github genesis] to its internal database.

- [What is a Feeler Connection? When is it used?]({{bse}}112247)
  User vnprc explains the purpose of Bitcoin Core's [feeler

>   connection][chaincode p2p] which is an outbound connection separate from the

Reviewer: ```suggestion
  connection][chaincode p2p] which is a temporary outbound connection separate from the
```

  default 8 outbound connections and 2 blocks-only outbound connections. The
  feeler connection is used to test potential new peers suggested from the

>   gossip network as well as test previously unreachable peers and evict them accordingly.

Reviewer: ```suggestion
  gossip network as well as test previously unreachable peers which are candidates for eviction.
```

- [Are OP_RETURN transactions not stored in chainstate database?]({{bse}}112312)
  Antoine Poinsot points out that since `OP_RETURN` outputs are
  [unspendable][bitcoin github unspendable], they are not stored in the

>   [chainstate directory][bitcoin wiki data].

Reviewer: Would the Bitcoin Core files doc be better than wiki? It is more accurate and closer to the source.
https://github.com/bitcoin/bitcoin/blob/master/doc/files.md#data-directory-layout
